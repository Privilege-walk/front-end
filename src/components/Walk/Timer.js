import React, { useState, useEffect } from "react";

import Box from '@mui/material/Box';

export default function Timer({questionIndex}) {

    const [startTime, setStartTime] = useState(Date.now());
    const [minutes, setMinutes] = useState(0);
    const [seconds, setSeconds] = useState(0);

    useEffect(() => {
        let interval = setInterval(() => {
            let ns = 60
            if (seconds == (ns-1)) {
                setMinutes((minutes + 1) % ns)
            }
            // setSeconds((seconds + 1) % ns)
            setSeconds(parseInt((Date.now() - startTime)/1000) % ns)
        }, 1000);
        return () => clearInterval(interval);
    })

    useEffect(() => {
        setStartTime(Date.now());
        setSeconds(0);
        setMinutes(0);
    }, [questionIndex])

    return (
        <Box component="span" sx={{ m: '20px 0px' }}>
            Time - {minutes < 10 ? ('0'+minutes) : minutes} : {seconds < 10 ? ('0'+seconds) : seconds}
        </Box>
    );
}