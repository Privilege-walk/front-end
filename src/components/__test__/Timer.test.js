import React from "react";
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import Timer from '../Walk/Timer';

test('renders without crashing',  () => {
    jest.useFakeTimers();
    let utils;
    act (() => {
        utils = render( <Timer questionIndex={0} />);
    }); 
    jest.advanceTimersByTime(100000);
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
})