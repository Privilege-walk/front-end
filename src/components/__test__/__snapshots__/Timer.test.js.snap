// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <span
    class="MuiBox-root css-msn52r"
  >
    Time - 01 : 39
  </span>
</DocumentFragment>
`;
